### ISSUE TITLE
PreDispatch tracing doesn't produce correct "val" for Sparse Fake tensors.

### ISSUE THREAD

**Issue by tugsbayasgalan** [pytorch/pytorch#124328]
### 🐛 Describe the bug

```python
from torch._functorch.aot_autograd import aot_export_module

class M(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, x):
        return (torch.ops.aten._to_sparse.sparse_dim(x, 1),)

gm, _ = aot_export_module(M(), [torch.randn(5, 5),], trace_joint=False, pre_dispatch=True)
for node in gm.graph.nodes:
    if node.op == "call_function":
        print(node.meta["val"] if "val" in node.meta else None)
# prints 
tensor(indices=tensor(..., size=(1, 0)),
       values=tensor(..., size=(0, 5)),
       device='meta', size=(5, 5), layout=torch.sparse_coo)
```

### Versions

main

**Comment by tugsbayasgalan** [comment]
Seems it happens when we call tensor.detach on fake sparse tensor.
